cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogDataOffsetCallback.java

```diff
@@ -44,8 +46,20 @@ public Object encodeSend(LogEvent logEvent) {
     String className = "";
     String simpleClassName = "";
     int lineNumber = 0;
+    Level level = event.getLevel();
+    StackTraceElement first = null;
     if (stackTraceElements.length > 0) {
-      StackTraceElement first = stackTraceElements[0];
+      first = stackTraceElements[0];
+      // If the line number from caller data is -1 and the event level is ERROR,
+      // use the line number from throwable proxy
+      if (first.getLineNumber() == -1 && level.equals(Level.ERROR)) {
```

Comment:
Why this is specific to the `ERROR` log level? What if someone log the stacktrace with lesser level (say `WARN`)?